lib/pos_fee_estimator.go

```diff
@@ -194,5 +222,149 @@ func (posFeeEstimator *PoSFeeEstimator) mempoolFeeEstimate(
 	priorityPercentileBasisPoints uint64,
 	maxBlockSize uint64,
 ) (uint64, error) {
-	panic("TODO: IMPLEMENT ME")
+	txnFee, err := posFeeEstimator.estimateTxnFeeGivenTransactionRegister(txn, posFeeEstimator.mempoolFeeEstimatorMempool.txnRegister, congestionFactorBasisPoints, priorityPercentileBasisPoints, maxBlockSize)
+	if err != nil {
+		return 0, errors.Wrap(err, "mempoolFeeEstimate: Problem computing txn fee")
+	}
+	return txnFee, nil
+}
+
+func computeFeeGivenTxnAndFeeRate(txn *MsgDeSoTxn, feeRateNanosPerKB uint64) (uint64, error) {
+	// Create a clone of the txn, so we don't modify the original.
+	txnClone, err := txn.Copy()
+	if err != nil {
+		return 0, errors.Wrap(err, "computeFeeGivenTxnAndFeeRate: Problem copying txn")
+	}
+	// Set the nonce to the maximum value if it's not already set.
+	if txnClone.TxnNonce == nil {
+		txnClone.TxnNonce = &DeSoNonce{
+			ExpirationBlockHeight: math.MaxUint64,
+			PartialID:             math.MaxUint64,
+		}
+	}
+	if txnClone.TxnNonce.ExpirationBlockHeight == 0 {
+		txnClone.TxnNonce.ExpirationBlockHeight = math.MaxUint64
+	}
+	if txnClone.TxnNonce.PartialID == 0 {
+		txnClone.TxnNonce.PartialID = math.MaxUint64
+	}
+
+	// Set the TxnFeeNanos to the maximum value.
+	txnClone.TxnFeeNanos = math.MaxUint64
+	txnFeeNanos, err := computeFeeRecursive(txnClone, feeRateNanosPerKB)
+	if err != nil {
+		return 0, errors.Wrap(err, "computeFeeGivenTxnAndFeeRate: Problem computing fee rate recursively")
+	}
+	return txnFeeNanos, nil
+}
+
+func computeFeeRecursive(txn *MsgDeSoTxn, feeRateNanosPerKB uint64) (uint64, error) {
```

Comment:
Add a comment for this function signature just to sketch out what this is intended for, what it computes, what the base case and recursive cases are. I had a bit of a hard time understanding it at first, so this will be helpful to future reviewers